### DISCUSSION TITLE
Ability to filter/suppress warning messages?

### DISCUSSION THREAD

**Discussion by drwestco** [maplibre/maplibre-gl-js#1313] [Ideas]
Specifically, the `Image "xxx" could not be loaded.` message in response to the `styleimagemissing` event. I'm fairly sure there are no critical side effects of ignoring this event - depending on style properties, the corresponding symbol may be omitted or only partially rendered until the image is available. We're intentionally delay-loading images in response to this event. While we can suppress the message by calling addImage with a dummy, empty image until the real image is available, this has negative perf implications - duplicate render calls, incorrect display until real image is available, etc.

I don't know if there are other warnings that are similarly ignorable. The ones about invalid properties in style tend to be more catastrophic..

Since MapLibre warns by directly calling `console.log`, we don't have a good way of suppressing these. A property on the `Map` class to invoke a callback for logging would be nice to have.

**Comment by HarelM** [comment]
I generally agree that a logger of some sort can be a good injectable properly, not to the map object, but maybe to the global maplibregl object.
But in theory, you could rewrite console.log, couldn't you?
